modules/cij/runner.py

```diff
@@ -372,35 +366,24 @@ def trun_to_file(trun, fpath=None):
 def trun_to_junitfile(trun, fpath=None):
     """Generate jUNIT XML from testrun YML"""
 
-    # pylint: disable=too-many-locals
-
     try:
         if fpath is None:
             fpath = junit_fpath(trun.conf["OUTPUT"])
 
         doc = minidom.Document()
-
         doc_testsuites = doc.createElement('testsuites')
 
-        duration = 0
-        if trun.stamp:
-            stamp_begin = trun.stamp.get("begin", None)
-            if not stamp_begin:
-                stamp_begin = time.time()
-
-            stamp_end = trun.stamp.get("end", None)
-            if not stamp_end:
-                stamp_end = time.time()
-
-            if stamp_end > stamp_begin:
-                duration = stamp_end - stamp_begin
+        stamp = trun.stamp
+        duration = max(
+            0,
+            stamp.get("end", time.time()) - stamp.get("begin", time.time())
+        )
```

Comment:
We can skip the if-statement altogether because the dataclass promises us that `trun.stamp` always contains a dict.

Another variation of the above is as follows, but requires the pylint-disable:

```python
        stamp_begin = trun.stamp.get("begin", time.time())
        stamp_end = trun.stamp.get("end", time.time())
        duration = max(0, stamp_end - stamp_begin) 
```

Reply:
Ah, what you did here seems nice, what I proposed was wrong... since yeah... it should not alter the trun, just derive the duration.

Reply:
How about:
```
duration = max(
    0,
    trun.stamp.get("end", time.time()) - trun.tamp.get("begin", time.time())
)
```

Ain't that a winner?

Reply:
Or... just:
```
trun.stamp["begin"] = trun.stamp.get("begin", time.time())
trun.stamp["end"] = trun.stamp.get("end", time.time())
```
if it makes sense to update the trun...